.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,101 @@
+          - os: macos-13 # for macosX64
+            additional-args: '-x jvmBenchmark -x jsBenchmark'
+          - os: windows-latest
+            additional-args: '-x jvmBenchmark -x jsBenchmark'
+    name: Performance regression check on ${{ matrix.os }} runner
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v4
+    - name: 'Set up JDK'
+      uses: 'actions/setup-java@v4'
+      with:
+        java-version: '11'
+        distribution: 'zulu'
+        cache: 'gradle'
+    - name: Validate Gradle Wrapper
+      uses: gradle/actions/wrapper-validation@v3
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@v3
+      with:
+        gradle-version: wrapper
+    - name: Run benchmarks
+      run: ./gradlew -p snake-kmp-benchmarks benchmark --no-parallel ${{ matrix.additional-args }}
+    - uses: actions/upload-artifact@v4
+      with:
+        name: bench-results-${{ matrix.os }}
+        path: ${{ env.BENCHMARK_RESULTS }}/main/**/*.json
+  collect-benchmarks-results:
+    runs-on: ubuntu-latest
+    needs:
+      - run-benchmark
+    permissions:
+      # deployments permission to deploy GitHub pages website
+      deployments: write
+      # contents permission to update benchmark contents in gh-pages branch
+      contents: write
+      # pull-requests permission to create comments on PR in case of alert
+      pull-requests: write
+    env:
+      RESULTS_DIR: bench-results
+    steps:
+    # without checkout step 'benchmark-action/github-action-benchmark' action won't work
+    - uses: actions/checkout@v4
+    - name: Download benchmark results
+      uses: actions/download-artifact@v4
+      with:
+        pattern: bench-results-*
+        path: ${{ env.RESULTS_DIR }}
+        merge-multiple: true
+    - name: Prepare and join benchmark reports
+      id: prep
+      run: |
+        for report in $(find ./${{ env.RESULTS_DIR }} -type f -name "*.json")
+        do
+          file_name=$(basename "$report")
+          platform="${file_name%.*}"
+          # Trim 'it.krzeminski.snakeyaml.engine.kmp.benchmark.' to make benchmark name more readable
+          jq "[ .[] | .benchmark |= \"${platform}.\" + ltrimstr(\"it.krzeminski.snakeyaml.engine.kmp.benchmark.\") | .params |= map_values(. |= match(\"data.+\"; \"g\").string) ]" $report > ${{ env.RESULTS_DIR }}/$platform.json
+        done
+        AGGREGATED_REPORT=aggregated.json
+        # Joined reports looks like this: [[{},{}], [{},{}]]
+        # We need to transform them into this: [{},{}]
+        ls ${{ env.RESULTS_DIR }}/*.json
+        jq -s '[ .[] | .[] ]' ${{ env.RESULTS_DIR }}/*.json > $AGGREGATED_REPORT
+        echo "report=$AGGREGATED_REPORT" >> $GITHUB_OUTPUT
+    - name: Store benchmark result
+      uses: benchmark-action/github-action-benchmark@v1
+      with:
+        name: SnakeKMP benchmarks
+        tool: 'jmh'
+        output-file-path: ${{ steps.prep.outputs.report }}
+        comment-on-alert: true
```

Comment:
@OptimumCode if the token issue is still blocking us, how about setting `comment-on-alert` to false for now, and relying on the fact that the job should fail?

Reply:
I have no objections to that. But I think if the secret with PAT is not visible in the workflow we won't be able to clone repo with results and push new results back. And this is the main issue here at the moment

Reply:
True! This problem seems to be with GitHub Actions, not this particular action. I'll play with this, e. g. by creating different kinds of secrets, although I think the action secret should work here.

Reply:
Me too. Probably it is happening because I am not a contributor - probably GitHub doesn't expose secrets in such cases. Or maybe because this is a PR from a fork. I will also dig into this

Reply:
> Or maybe because this is a PR from a fork

Oh, right! Let me make you a contributor, then you can move your branch to this repo and perhaps recreate the PR. Let's try it!

Reply:
Thank you! Once I have access to the computer I will move the branch.

RE secrets: this is what I have found

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow

Probably, it can be solved with an environment but I am not sure

Reply:
Looks like this might be a solution: https://github.com/community/community/discussions/50161#discussioncomment-5824158.

If we don't configure the execution that works for PRs from forks there is not much point in having this job, I think... I will do required changes in the workflow. @krzema12 could you please create an environment that I could use in the workflow? Also could you please add this PAT to the environment you create?

Reply:
The environment is just a safety measure

Reply:
Actually, on a second thought, we might not need an environment... I think we can split the workflow into two separate workflows:
- One runs benchmark and stores results as artifacts
- another downloads results and executes benchmark action (which requires access to secrets). This can be done using **workflow_run** trigger

As a result, code from pr is executed in safe environment without access to secrets and results processing is executed with access to secrets

Reply:
@krzema12 what do you think about this idea? It sounds complicated but as a result, provides a bit more safety

Reply:
How about we go with the current approach, as it already gives us some value when making changes to the library (right now it's mostly @aSemy who e. g. does some refactoring), and if the volume of external contributions rise, we'll consider splitting the workflow like you propose?

Reply:
This is completely up to you) In this case, I will move the branch and recreate the PR